README-for-review.md

```diff
@@ -0,0 +1,384 @@
+
+###EntityType
+
+Actually, you will find no definition of `Entity` in the library. Repositories use entities as an associated type (`EntityType`); the only restriction is that any entity you create should conform to these three protocols:
+
+- `Identifiable`
+  - To allow the entity to be unequivocally identified via an `id` field.
+- `DictionaryInitializable`
+  - To allow the entity to be initialized from a dictionary (parsed in).
+- `DictionaryRepresentable`
+  - To allow the entity to be represented through a dictionary (parsed out).
+
+
+
+### Identifiers
+
+Jayme takes a flexible approach regarding identifiers. As of Jayme 2.0, identifiers are not tied to any concrete type; it's up to you to define which kind of identifier each of your entity types will use. 
+
+You could have entities having `Int` ids, other entities having `String`, or all of your them having the same identifier type. 
+
+You can also use your own identifier types. This is particularly useful when you have to face complex scenarios. For instance, you might need to deal with *local identifiers* vs. *server identifiers*, and you might encounter several approaches for doing so. It's recommendable then that you take a look at the example project to see a complex identifier implementation (check out `PostIdentifier`).
+
+
+### Jayme Default Core
+
+Whereas `Repository` and `Backend` definitions are only contracts, Jayme includes some structures based on them, which include default implementations (hence behaviors) that are based on conventions and standards that we normally follow at [Inaka](http://inaka.net/).
+
+These structures are:
+
+- `NSURLSessionBackend`: A *class* that connects to a server using `NSURLSession` mechanisms.
+- `CRUDRepository`: A *protocol* that provides elemental CRUD functionality.
+- `PagedRepository`: A *protocol* that provides read functionality with pagination.
+
+Normally you will use these. However, if you need, you can ignore them and implement your own common behaviors by conforming directly to the base `Repository` and `Backend` contracts, as shown below:
+
+![Jayme's Customization](https://raw.githubusercontent.com/inaka/Jayme/master/Assets/architecture-diagram-2.png)
+
+Next, there's a detailed description of each item mentioned above:
+
+#### NSURLSessionBackend
+
+- This class provides a backend that connects to a server using HTTP REST requests via `NSURLSession`.
+
+
+- By default, it connects to `localhost:8080`, but you can change the base URL path by passing in a custom `NSURLSessionBackendConfiguration` object to its initializer. 
+- You can also set custom HTTP headers to be used in the requests when initilizing your own `NSURLSessionBackendConfiguration` instance.
+- You can also customize the `NSURLSession` and `HTTPResponseParser` objects that are asked upon `NSURLSessionBackend` initialization. However, doing so is discouraged; these last two parameters have been purely added there for unit-testing purposes.
```

Comment:
'doing so is discouraged.' What you mean here is that you do not recommend customizing those objects? If so, we need to change the sentence.